### ISSUE TITLE
defaultSeverity from shareable config doesn't work

### ISSUE THREAD

**Issue by guoyunhe** [stylelint/stylelint#5858]
### What steps are needed to reproduce the bug?

```
git clone git@github.com:guoyunhe/test-stylelint-default-severity.git
cd test-stylelint-default-severity
npm install
npm test
```

The test repository depends on a shareable configuration `eslint-config-ali`, which has `defaultSeverity: "warning"`.

### What Stylelint configuration is needed to reproduce the bug?

```json
{ "extends": "stylelint-config-ali" }
```

### How did you run Stylelint?

npm test

### Which version of Stylelint are you using?

14.2.0

### What did you expect to happen?

Stylelint should report warning:

```
index.css
 2:10  ⚠  Expected "#ffffff" to be "#fff"  color-hex-length
```

### What actually happened?

Stylelint reported error:

```
index.css
 2:10  ✖  Expected "#ffffff" to be "#fff"  color-hex-length
```

**Comment by guoyunhe** [comment]
Sorry, its the configuration's problem.
